From d0d61479c51b09d2286039875c8be1a515b17cd1 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:48 -0400
Subject: [tools] perf: Add support to dynamically get cacheline size

Message-id: <1410181752-25631-284-git-send-email-jolsa@redhat.com>
Patchwork-id: 91925
O-Subject: [PATCH RHEL7.1 BZ1134356 283/307] perf tools: Add support to dynamically get cacheline size
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 2b1b71003ea809e619bd73e74dfc2a73069de66f
Author: Don Zickus <dzickus@redhat.com>
Date: Fri May 30 16:10:05 2014 -0400

description
===========
Different arches may have different cacheline sizes.  Look it up and set
a global variable for reference.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/perf.c b/tools/perf/perf.c
index 78f7b92..95c58fc 100644
--- a/tools/perf/perf.c
+++ b/tools/perf/perf.c
@@ -458,6 +458,7 @@ int main(int argc, const char **argv)
 
  /* The page_size is placed in util object. */
  page_size = sysconf(_SC_PAGE_SIZE);
+ cacheline_size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE);
 
  cmd = perf_extract_argv0_path(argv[0]);
  if (!cmd)
diff --git a/tools/perf/util/util.c b/tools/perf/util/util.c
index 7fff6be..95aefa7 100644
--- a/tools/perf/util/util.c
+++ b/tools/perf/util/util.c
@@ -17,6 +17,7 @@
  * XXX We need to find a better place for these things...
  */
 unsigned int page_size;
+int cacheline_size;
 
 bool test_attr__enabled;
 
diff --git a/tools/perf/util/util.h b/tools/perf/util/util.h
index b03da44..6686436 100644
--- a/tools/perf/util/util.h
+++ b/tools/perf/util/util.h
@@ -304,6 +304,7 @@ char *rtrim(char *s);
 void dump_stack(void);
 
 extern unsigned int page_size;
+extern int cacheline_size;
 
 void get_term_dimensions(struct winsize *ws);
 
-- 
1.7.1